config.c

```diff
@@ -1911,19 +1911,78 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                                     S_ISDIR(sb_glob.st_mode)) {
                                 continue;
                             }
 
+                            /* Resolves the directory but not the filename, to be able to detect symlinks */
+                            copy = strdup(globResult.gl_pathv[glob_count]);
+                            if (copy == NULL) {
+                                message_OOM();
+                                logerror = 1;
+                                goto duperror;
+                            }
+                            /* Because dirname() modifies the path, order is important */
+                            baseName = basename(copy);
+                            dirName = dirname(copy);
+                            resolvedDir = realpath(dirName, NULL);
+                            if (!resolvedDir) {
+                                if (errno == ENOENT) {
+                                    /* We don't know yet if missingok... */
+                                    resolvedDir = strdup(dirName);
+                                    if (resolvedDir == NULL) {
+                                        message_OOM();
+                                        logerror = 1;
+                                        goto duperror;
+                                    }
+                                } else if (errno == ENOMEM) {
+                                    message_OOM();
+                                    logerror = 1;
+                                    goto duperror;
+                                } else {
+                                    message(MESS_ERROR,
+                                        "%s:%d resolve error on log entry for %s: %s\n",
+                                        configFile, lineNum, globResult.gl_pathv[glob_count],
+                                        strerror(errno));
+                                    logerror = 1;
+                                    free(copy);
+                                    goto duperror;
+                                }
+                            }
+                            resolved_path = (char *)malloc(strlen(resolvedDir) + 1 + strlen(baseName) + 1);
+                            if (resolved_path == NULL) {
+                                message_OOM();
+                                logerror = 1;
+                                goto duperror;
+                            }
+                            (void) sprintf(resolved_path, "%s/%s", resolvedDir, baseName);
+                            free(resolvedDir);
+                            free(copy);
+
+                            /* Skip duplicates for the current set */
+                            for (k = 0; k < newlog->numFiles; k++) {
+                                if (!strcmp(newlog->files[k], resolved_path)) {
+                                    message(MESS_DEBUG,
```

Comment:
Is this worth `MESS_WARN`?

Reply:
I don't think it worth `MESS_WARN` because it's not an issue as such.